buildSrc/src/main/java/org/opensearch/gradle/pluginzip/Publish.java

```diff
@@ -9,83 +9,71 @@
 
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
-import org.gradle.api.logging.Logger;
-import org.gradle.api.logging.Logging;
 import org.gradle.api.publish.PublishingExtension;
 import org.gradle.api.publish.maven.MavenPublication;
-import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
 
 import java.nio.file.Path;
 import org.gradle.api.Task;
 
 public class Publish implements Plugin<Project> {
 
-    private static final Logger LOGGER = Logging.getLogger(Publish.class);
-
-    public final static String EXTENSION_NAME = "zipmavensettings";
+    // public final static String EXTENSION_NAME = "zipmavensettings";
+    // public final static String PLUGIN_ZIP_PUBLISH_POM_TASK = "generatePomFileForPluginZipPublication";
+    // public final static String LOCALMAVEN = "publishToMavenLocal";
+    // TODO: we can remove these ^^
     public final static String PUBLICATION_NAME = "pluginZip";
     public final static String STAGING_REPO = "zipStaging";
-    public final static String PLUGIN_ZIP_PUBLISH_POM_TASK = "generatePomFileForPluginZipPublication";
-    public final static String LOCALMAVEN = "publishToMavenLocal";
     public final static String LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";
-    public String zipDistributionLocation = "/build/distributions/";
+    // TODO: the path ^^ needs to use platform dependant file separators
 
-    public static void configMaven(Project project) {
+    private boolean isZipPublicationPresent(Project project) {
+        PublishingExtension pe = project.getExtensions().findByType(PublishingExtension.class);
+        if (pe == null) {
+            return false;
+        }
+        return pe.getPublications().findByName(PUBLICATION_NAME) != null;
+    }
+
+    private void addLocalMavenRepo(Project project) {
         final Path buildDirectory = project.getRootDir().toPath();
-        project.getPluginManager().apply(MavenPublishPlugin.class);
+        // project.getPluginManager().apply(MavenPublishPlugin.class);
```

Comment:
@prudhvigodithi @reta Is this line really needed?

Reply:
It looks this this the way to stack plugins, at least I see similar approach here [1]


[1] https://github.com/gradle/gradle/blob/master/subprojects/maven/src/main/java/org/gradle/api/publish/maven/plugins/MavenPublishPlugin.java#L95

Reply:
You are correct @reta , I fixed this.

In fact I went one step further and used apply() for two plugins:
```java
project.getPluginManager().apply("nebula.maven-base-publish");
project.getPluginManager().apply(MavenPublishPlugin.class);
```
This change allowed me to remove all uses of the nebula plugin from testing gradle.project files (those that are used in `PublishTests.java`). Since the Nebula plugin is really used very often by the MavenPublishPlugin I think it makes sense to directly include it programatically. (Any risks that I might be missing?)

Because I really wanted to be sure what this change caused I added two new unit tests  –`applyZipPublicationPluginNoConfig` and `applyZipPublicationPluginWithConfig` – to verify what kind of gradle tasks are present in the project when `opensearch.pluginzip` plugin is used. These were a bit tricky to implement but I think they are valuable.

Finally, I introduced a `warn` message for the case when the plugin is used but no publication is configured for it. Unit test provided as well.